CONTRIBUTING.md

```diff
@@ -0,0 +1,412 @@
+  // ... 
+}
+
+List.Item = ListItem
+```
+
+```js
+import ListItem from './ListItem'
+
+class List {
+  static Item = ListItem
+}
+````
+
+### Testing
+
+Run tests during development with `npm run test:watch` to re-run tests on file changes.
+
+#### Common Tests
+
+There are many common things to test for.  Because of this, we have [`test/specs/commonTests.js`][1].
+
+>This list is not updated, check the [source][1] for current tests and inline documentation.
+
+```js
+common.isConformant()
+common.hasUIClassName()
+common.hasSubComponents()
+common.isTabbable()
+common.rendersChildren()
+common.implementsAlignedProp()
+common.implementsIconProp()
+common.implementsImageProp()
+common.propKeyOnlyToClassName()
+common.propValueOnlyToClassName()
+common.propKeyAndValueToClassName()
+common.propKeyOrValueToClassName()
+````
+
+##### Usage
+  
+Every common test receives your component as its first argument.
+
+```js
+import React from 'react'
+import * as common from 'test/specs/commonTests'
+import Menu from 'src/collections/Menu/Menu'
+import MenuItem from 'src/collections/Menu/MenuItem'
+
+describe('Menu', () => {
+  common.isConformant(Menu)
+  common.hasUIClassName(Menu)
+  common.rendersChildren(Menu)
+  common.hasSubComponents(Menu, [MenuItem]) // some take additional arguments
+})
+```
+
+The last argument to a common test is always `requiredProps`.  If your component requires certain props to render, you can pass those last: 
+
+```js
+import * as common from 'test/specs/commonTests'
+import Select from 'src/addons/Select/Select'
+
+const requiredProps = {
+  options: [],
+}
+
+describe('Select', () => {
+  common.isConformant(Select, requiredProps)
+})
+```
+
+##### isConformant (required)
+
+This is the only required test.  It ensures a consistent baseline for the framework, and can can help you get your component off the ground.
+
+>This list is not updated, check the [source][1] for the latest assertions.
+
+1. The [static `_meta`](#_meta) object is valid
+1. Component and filename are correct
+1. Events are properly handled
+1. Extra `props` are spread
+1. Base `className`s are applied
+1. Component is exported if public / hidden if private
+
+### State
+
+TODO
+
+#### AutoControlledComponent
+
+TODO
+
+## Documentation
+
+Our docs are generated from doc block comments, `propTypes`, and hand written examples. 
+
+### Website
+
+Developing against the doc site is good way to try your component as you build it. Run the doc site with:
+
+```sh
+npm start
+```
+
+### Components
+
+A doc block should appear above a component class or function to describe it:
+
+```js
+/**
+ * A <Select /> is sugar for <Dropdown selection />.
+ * @see Dropdown
+ */
+function Select(props) {
+  return <Dropdown {...props} selection />
+}
+```
+
+### Props
+
+A doc block should appear above each prop in `propTypes` to describe them:
+
+>Limited props shown for brevity.
+
+```js
+Label.propTypes = {
+  /** A label can reduce its complexity */
```

Comment:
All other descriptions have periods, so since you have to edit this file anyways, would be nice to make it consistent.